### PR TITLE
Update notifications for assets & generated files

### PR DESCRIPTION
**Pull request by krasnotsvetov**
1. Show "Show in Unity" only when connection is available
2. Show "Edit corresponding .asmdef file" for .csproj with asmdefs.

### CODE REVIEW

File: resharper/resharper-unity/src/AsmDefNew/Psi/Caches/AsmDefNameCache.cs [hunk 1]
```diff
@@ -14,20 +16,25 @@
...
 using JetBrains.ReSharper.Psi.Resolve;
 using JetBrains.Util;
+using JetBrains.Util.Collections;
 
 namespace JetBrains.ReSharper.Plugins.Unity.AsmDefNew.Psi.Caches
 {
     [PsiComponent]
     public class AsmDefNameCache : SimpleICache<AsmDefCacheItem>
     {
+        private readonly IShellLocks myShellLocks;
```
Comment (citizenmatt) [inline resharper/resharper-unity/src/AsmDefNew/Psi/Caches/AsmDefNameCache.cs]: Unused field?

File: resharper/resharper-unity/src/Rider/UnityHost.cs [hunk 2]
```diff
@@ -3,8 +3,11 @@
 using JetBrains.Application.Threading.Tasks;
 using JetBrains.Lifetimes;
 using JetBrains.ProjectModel;
+using JetBrains.Rd.Tasks;
 using JetBrains.ReSharper.Host.Features;
+using JetBrains.ReSharper.Plugins.Unity.AsmDefNew.Psi.Caches;
 using JetBrains.ReSharper.Plugins.Unity.Feature.Caches;
+using JetBrains.ReSharper.Resources.Shell;
```
Comment (citizenmatt) [inline resharper/resharper-unity/src/Rider/UnityHost.cs]: Unused extra `using`s?

File: resharper/resharper-unity/src/Rider/Notifications/GeneratedFileNotification.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        public GeneratedFileNotification(Lifetime lifetime, UnityHost unityHost, ConnectionTracker connectionTracker, UnityEditorProtocol editorProtocol, ISolution solution,
+            AsmDefNameCache asmDefNameCache, [CanBeNull] TextControlHost textControlHost = null, [CanBeNull] SolutionLifecycleHost solutionLifecycleHost = null,
+            [CanBeNull] NotificationPanelHost notificationPanelHost = null)
+        {
+            if (solutionLifecycleHost == null)
+                return;
+            
+            myNotificationPanelHost = notificationPanelHost;
+                
+            solutionLifecycleHost.FullStartupFinished.Advise(lifetime, _ =>
```
Comment (citizenmatt) [inline resharper/resharper-unity/src/Rider/Notifications/GeneratedFileNotification.cs]: These subscriptions will happen for non-Unity projects. Add a check for Unity solution somewhere?

File: resharper/resharper-unity/src/Rider/Notifications/GeneratedFileNotification.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                                    {
+                                        unityHost.PerformModelAction(t => t.AllowSetForegroundWindow.Start(unityStateLifetime, Unit.Instance).Result.Advise(unityStateLifetime,
+                                            __ =>
+                                            {
+                                                editorProtocol.UnityModel.Value?.ShowFileInUnity.Fire(strPath);
+                                            }));
+                                    }));
+                            }
+                        
+                            notificationPanelHost.AddNotificationPanel(unityStateLifetime, host, new NotificationPanel("This file is generated by Unity. Any changes made will be lost.", "UnityGeneratedFile", elements.ToArray()));
```
Comment (citizenmatt) [inline resharper/resharper-unity/src/Rider/Notifications/GeneratedFileNotification.cs]: It looks like we lose the "This file is generated by Unity" message if there is no connection. We should still show it, even if Unity isn't running.

Comment (krasnotsvetov) [reply resharper/resharper-unity/src/Rider/Notifications/GeneratedFileNotification.cs]: No, we will hide hyperlink only

**Pull request merged**